web_src/js/features/repo-issue.js

```diff
@@ -380,6 +381,10 @@ export async function handleReply($el) {
   form.removeClass('gt-hidden');
 
   const $textarea = form.find('textarea');
+  const $dropzone = form.find('.dropzone')[0];
+  if (!$dropzone.dropzone) {
+    initDropzone($dropzone);
```

Comment:
I think this line should be moved into `if (!editor) {` (line 389)

That `if` blocks means there is no editor/uploader, it will initialize new ones.

Reply:
When I move it inside the `if (!editor) {` without checking whether the dropzone is already initialized, it results in an error "Uncaught (in promise) Error: Dropzone already attached.". So it's somehow in a state where the editor has not been initialized yet, but the dropzone has.

But if I remove the dropzone initialization from this function entirely, I get this uninitialized dropzone when replying twice:

![image](https://github.com/go-gitea/gitea/assets/2212615/4cabeef8-0bc5-4e6b-9ffb-b0d74cf3972d)

Reply:
🙏。It seems that there are some underlying bugs. I will take a look.

Reply:
Hmm, I think I get the root problem now.

At the beginning, the "dropzone" is initialized by the "global init", but the editor is not initialized.

When you submit and edit again, the HTML is partially reloaded, so none of them is initialized ....

Will try to figure out how to handle case. For the worst case, I'd like to leave a comment there.

Reply:
Does this change look good to you?

```diff
diff --git a/web_src/js/features/repo-issue.js b/web_src/js/features/repo-issue.js
index 47b395fb3f..10faeb135d 100644
--- a/web_src/js/features/repo-issue.js
+++ b/web_src/js/features/repo-issue.js
@@ -381,12 +381,13 @@ export async function handleReply($el) {
   form.removeClass('gt-hidden');
 
   const $textarea = form.find('textarea');
-  const $dropzone = form.find('.dropzone')[0];
-  if (!$dropzone.dropzone) {
-    initDropzone($dropzone);
-  }
   let editor = getComboMarkdownEditor($textarea);
   if (!editor) {
+    // FIXME: the initialization of the dropzone is not consistent.
+    // When the page is loaded, the dropzone is initialized by initGlobalDropzone, but the editor is not initialized.
+    // When the form is submitted and partially reload, none of them is initialized.
+    const dropzone = form.find('.dropzone')[0];
+    if (!dropzone.dropzone) initDropzone(dropzone);
     editor = await initComboMarkdownEditor(form.find('.combo-markdown-editor'));
   }
   editor.focus();
```

Reply:
Yes, applied. Thanks.